### DISCUSSION TITLE
Descriptions dropped from scalars

### DISCUSSION THREAD

**Discussion by jonathanstanley** [dotansimha/graphql-code-generator#8327] [Q&A] [answered]
today my scalars lost all their descriptions in generated typescript

ex:
```diff
/** My Scalars */
export type Scalars = {
- /* A UUID*/
  ID: string;

- /** A link location*/
  Url: string;
};
```

It has been generating just fine for a while. I don't think this is a local issue because the server graphql hasn't changed. I believe this is a bug because it just started after I updated codegen. Any ideas to fix?

```yml
    plugins:
      - typescript
      - typescript-operations
      - typescript-graphql-request
```

relevant package.json:
```json
{
    "@graphql-codegen/add": "^3.1.1",
    "@graphql-codegen/cli": "^2.6.2",
    "@graphql-codegen/typescript": "^2.4.8",
    "@graphql-codegen/typescript-graphql-request": "^4.4.8",
    "@graphql-codegen/typescript-operations": "^2.3.5",
}
```

**Comment by charlypoly** [comment]
Hi @jonathanstanley, from which version of each codegen package were you updating from?

This would greatly help me narrow down the root issue, thanks!

**Reply by jonathanstanley** [reply]
> Hi @jonathanstanley, from which version of each codegen package were you updating from?

Thanks! Unfortunately, I'm not quite sure since I did several package updates without needing to re-run the codegen. I tried reverting a couple versions and no help.  The closest starting point I could determine was the package.json I posted. It is a bit subtle to trace the history, and I understand if this means you are unable to resolve it.  I'm also hoping if anyone else finds this issue they might comment here to help track it down.

**Reply by charlypoly** [reply]
I will try to reproduce it and provide a patch.

--

Related issue: https://github.com/dotansimha/graphql-code-generator/discussions/8357

**Reply by charlypoly** [reply]
Hi again @jonathanstanley,

I could not reproduce your issue, cf: https://github.com/charlypoly/codegen-repros/blob/master/codegen-field-scalars-description/types.ts#L13

Could you provide an excerpt of your schema descriptions that fails that be translated to TS comments? 🙏🏼

**Reply by jonathanstanley** [reply]
Here is a reproduction: https://codesandbox.io/s/spring-tdd-qoxfwp?file=/types.ts

Notice the `Email` scalar does not have a description comment, but `Url` scalar does.

In creating this, I realized this could be caused by using `#` vs `"""` comment format (see https://codesandbox.io/s/spring-tdd-qoxfwp?file=/schema.graphql). So perhaps there was a recent change that dropped the ability to resolve `#` comments?

**Comment by charlypoly** [comment, marked as answer]
@jonathanstanley, after some research, found out that this breaking change has been introduced with the upgrade to `graphql-js^16`, which dropped support for `# [...]` descriptions in schemas (the GraphQL spec has also been updated on this matter)
The new spec states that:

- `# [...]` syntax is used for comments (which are not collected as descriptions)
- `" [...]"` and its multiline counterpart syntax should be used for descriptions (which are collected by codegen for TS comments)

**Reply by jonathanstanley** [reply]
thank you very much!  Here's a link for more info in case others have this issue: https://github.com/graphql/graphql-js/pull/2900#issuecomment-1255209795

**Reply by jonathanstanley** [reply]
After we checked this further, the `#` vs `"` issue is not the problem in our case. (although I thought it was)

for some reason, our schema was/is using `"` properly, however when I downloaded it from the playground, it seems the playground converted those descriptions into comments. so the reproduction I creates is not accurate.

However, I don't have much more information and this has me thinking it may be a local problem, so I will leave resolved unless/until I can create a reproduction. Thanks!
